rpc_api.go

```diff
@@ -154,19 +154,32 @@ func (rpcapi *ClusterRPCAPI) ID(ctx context.Context, in struct{}, out *api.ID) e
 	return nil
 }
 
-// Pin runs Cluster.Pin().
-func (rpcapi *ClusterRPCAPI) Pin(ctx context.Context, in *api.Pin, out *struct{}) error {
-	return rpcapi.c.Pin(ctx, in)
+// Pin runs Cluster.pin().
+func (rpcapi *ClusterRPCAPI) Pin(ctx context.Context, in *api.Pin, out *api.Pin) error {
+	// we do not call the Pin method directly since that method does not
+	// allow to pin other than regular DataType pins. The adder will
+	// however send Meta, Shard and ClusterDAG pins.
+	pin, _, err := rpcapi.c.pin(ctx, in, []peer.ID{})
```

Comment:
Most cluster method names match with rpc method names. This confuses, because there is a `Pin` method in cluster.

Instead of this we can support all types in `cluster.Pin` method. cluster.Pin is used by cluster.PinPath. So we would have to set pin options ( and thus pin type) in pinPathHandler(rest api). It already happens for pinHandler.

Reply:
pinType is not an option that should be set by the user. This is a good example where our internal API does not need to match our external one.

Reply:
sorry, I didn't mean it to be set by user.
I meant like this
https://github.com/ipfs/ipfs-cluster/blob/b6ba67804fc261446d32807660751c93df516377/api/rest/restapi.go#L1032

Reply:
pinType is not a `PinOptions`. I'm not sure what you mean.

Reply:
What I was proposing would require significant changes, PinPath rpc method accepting `types.Pin` instead of `types.PinPath` (in order to pass pin Type)

 We can let this be.